lib/rules/dont-allow-404-response.js

```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  meta: {
+    docs: {
+      description: 'do not allow 404 as expected response for network requests',
+    },
+    fixable: false,
+  },
+  create(context) {
+    return {
+      'AwaitExpression CallExpression ObjectExpression Property[key.name="expectedResponse"] > Literal[value=404]': (node) => {
+        context.report(node, "Use 204 instead of 404 status code to indicate no content");
```

Comment:
I think we also need to cover the hooks `errorCodes: [404],` case, e.g. in modules/landing/src/api/email-returning-user/email-returning-user-configuration.js

Reply:
This would be for `successCodes`, right? It should be fine to include it in `errorCodes`?

Reply:
I think we should avoid using 404 as an expected error code as well...We should be using 204 instead of 404 to indicate that not found or not content is a possible error scenario. I have updated the eslint rule to include more conditions

Reply:
Yep, what Chaitra said. We don't want to knowingly return 404s from the server at all.

Reply:
👍 makes sense to me.